Some role based access functionality

EvanShumaker opened the pull request:
# Summary
This PR adds backend/users/permissions.py to assist with role permissions / functionality.

# Description
I have added simple helper functions and permission classes to be used in other views or view sets. These can be called to verify if a user is of a certain role (i.e. professor, admin, or TA). It just needs to be imported from users.permissions.

The permission classes should allow for something similar using DRF's advanced filtering. In a view set you could write permission_classes = [..., IsAdmin]. This ensures only an admin can access the view. It serves a similar purpose as the helper function, but allows us to roll multiple permission classes into one without using a long 'if' statement.

There is also a logging function defined in permissions.py. This is for our use to keep track of which users do what action. Calls to this function should be made from inside a view, and it adds to the logger variable. We can write this out to a file if we want, but this isn't implemented yet.

# Documentation
Mentioned in the description, but all the helper functions and permission classes could be used in our views for role based access. The log function in that permissions.py file could be used there as well

# Unit Tests
4. No tests.

No tests yet, waiting for a potential db update before I create objects to make unit tests with.

# PR Reviews
Hoping @donessie94 and @stevenfojas670 could look at this, I believe they would be the most knowledgeable about this stuff.

EvanShumaker committed:
Implemented some role permissions and checking functions. Also a simple logging function if we want

EvanShumaker committed:
Updated permissions.py

EvanShumaker committed:
Added a permission class for Professor and TA check, since they're kinda the same

EvanShumaker committed:
Added an Enrollments table, meaning I made changes to admin, models, serializers, urls, and views

donessie94 commented:
These are great helper functions, I went trough the code and it looks great and makes sense. Good work

EvanShumaker committed:
Resolved merge conflicts

EvanShumaker committed:
Added an enrollments table. Meaning i changed models, serializers, urls, views, and admin.py

stevenfojas670 commented:
I couldn't run the backend. In courses/models you're referencing a student table that doesn't exist.

![image](https://github.com/user-attachments/assets/4dc8a475-336d-4a00-a7de-f9f4ed8dfec5)

EvanShumaker committed:
Update courses/models.py

Ive added an import in courses/models.py to include the Student table from the assignment app since I reference it in my enrollment table

stevenfojas670 reviewed (approved):
Backend runs without error and the code looks good. The enrollments table is a great implementation. Good stuff.

stevenfojas670 committed:
Added a gitignore

stevenfojas670 committed:
Merge branch 'dev' into dev
